test/AppNotificationTests/BaseTestSuite.cpp

```diff
@@ -0,0 +1,640 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "AppNotification-Test-Constants.h"
+#include "AppNotificationHelpers.h"
+#include "BaseTestSuite.h"
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Windows::ApplicationModel::Background;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::Management::Deployment;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::System;
+using namespace winrt::Microsoft::Windows::AppNotifications;
+using namespace AppNotificationHelpers;
+
+void BaseTestSuite::ClassSetup()
+{
+    ::Test::Bootstrap::SetupPackages();
+}
+
+void BaseTestSuite::ClassCleanup()
+{
+    ::Test::Bootstrap::CleanupPackages();
+}
+
+void BaseTestSuite::MethodSetup()
+{
+    ::Test::Bootstrap::SetupBootstrap();
```

Comment:
Why is this part of MethodSetup and not ClassSetup? Don't we want to do these test operations only once rather than every time a test runs which seems wasteful?

Reply:
Since TAEF launches a new process to test each method, that process needs to bootstrap instead of the parent process. This pattern is done by Howard as well.

Reply:
Can we do a quick experiment by moving this stuff to ClassSetup and ClassCleanup?

Reply:
Tested this and unpackaged tests fail.

Reply:
Coming back to this, it is ok to do one-time setup/cleanup in class functions.

Reply:
> Coming back to this, it is ok to do one-time setup/cleanup in class functions.

Sure. That's actually the _typical_ use case. Long as your test methods don't _alter_ what's done in setup.

Dynamic Dependencies and DeploymentManager are the exception cases, as they routinely work "before the runtime is (all or partly) installed". As long as AppNotifications isn't doing any 'runtime install/setup' work and just expects "the runtime is properly setup for use" as a prerequisite then you're golden.

One TAEF property to be aware of is if you set IsolationLevel=Method because you want your test methods each run in a new process. DynDep does this in some cases, 'cause, bootstrapping special weirdness. If you need this behavior then when/how your Setup/Cleanup fixtures run may also be...unusual. But this should all be rare birds.